### PR TITLE
Add support for generating fields/properties for property patterns.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/9090

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by gafter** [comment]
@CyrusNajmabadi This will have to wait until after #10866; I'm removing property patterns from the code base before insertion into future, and any changes to the pattern-matching code will cause merge heck.

**Comment by DustinCampbell** [comment]
@CyrusNajmabadi, @gafter: Should we close this ancient PR until we're ready for property patterns?

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi, @gafter: Should we close this ancient PR until we're ready for property patterns?

Works for me.  If we add property patterns back in someone can resurrect this :)

**Comment by CyrusNajmabadi** [comment]
@gafter Can you give me a set of interesting examples that demonstrate how people can use recursive patterns?  I want to make sure this features handles all the ones that make sense.  Thanks!

**Comment by gafter** [comment]
Yes, I will prepare a set of examples.

**Comment by CyrusNajmabadi** [comment]
Awesome, thanks.  It doesn't have to be huge.  Just like 1 (or 2) tops, for each interesting new construct.  And maybe some special 'oddities' (if they exist) to be aware of.

**Comment by gafter** [comment]
I will include some complete examples here in a moment, but just to point out some oddities first...

A property pattern checks that the input value is non-null, and that each named property matches the given pattern. But you can use no patterns if you just want a non-null check:

`if (o is {})`

The `switch` expression (parser and code gen) has some fixes pending that are not integrated yet, so it is broken in `features/recursive-patterns` at this moment, so better to focus on the switch statement and is-pattern expression.

**Comment by CyrusNajmabadi** [comment]
> ```if (o is {})```

Good case.  Just to verify, you should be able to write the following right:

```C3
if (o is { [|X|] is int i }
```

And this just means: "'o' is non-null and has a field/property 'X' that matches the pattern 'int i'"?

**Comment by CyrusNajmabadi** [comment]
@gafter ^

**Comment by CyrusNajmabadi** [comment]
Also, do we have a list of all patterns written down somewhere, i'd like to test out a variety of combinations.  What i know of off the top of my head:

1. Type pattern: ```int i```
2. Constant pattern: ```0```
3. Allow-null-of-any-type-pattern: ```var x```
4. non-null property pattern: ```{ ... }```
5. non-null + type + property pattern: ```X { ... }```
6. Discard pattern: ```_```

Did we do deconstruction patterns?  i.e. something like  ```SomeType(...)```?  If so, what syntax did we settle on for that?

**Comment by CyrusNajmabadi** [comment]
Also, inside a property-pattern, what syntax are the field/property pattern matchers?  At one point it was ```{ X is int i }```, is that still the case?  or did we go with another syntax that isn't just a list of "IsPatternExpression"s?

**Comment by gafter** [comment]
Yes, you've got that. In its most general form, a recursive pattern can be

`Type ( SubpatternList ) { PropertySubpatternList } Identifier`

You can omit any of the parts, except that you can't have only a type, only an identifier, or nothing.

**Comment by gafter** [comment]
See also https://github.com/dotnet/csharplang/issues/1054 and note that we added a shorthand discard pattern `_`.

File: src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs [hunk 1]
```diff
@@ -7606,6 +7606,44 @@ void Method()
...
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    void M2()
+    {
+        object o = null;
+        if (o is Blah { [|X|] is int i })
+        {
```
Comment (gafter) [inline src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: The syntax should be
``` c#
        if (o is Blah { [|X|]: int i })
```

File: src/Features/Core/Portable/GenerateMember/AbstractGenerateMemberService.cs [hunk 2]
```diff
@@ -154,6 +154,20 @@ protected AbstractGenerateMemberService()
                 isStatic = false;
                 return;
             }
+            else if (syntaxFacts.IsIsPatternExpressionExpression(expression) &&
+                     syntaxFacts.IsConstantPattern(expression.Parent.Parent) &&
+                     syntaxFacts.IsSubPatternElement(expression.Parent.Parent.Parent) &&
```
Comment (gafter) [inline src/Features/Core/Portable/GenerateMember/AbstractGenerateMemberService.cs]: You're testing for a a syntax error that is recovered in a particular way.

**Comment by gafter** [comment]
The syntax for inside a property pattern is currently a colon instead of `is`.

---
In reply to: [377326947](https://github.com/dotnet/roslyn/pull/10873#issuecomment-377326947) [](ancestors = 377326947)

**Comment by CyrusNajmabadi** [comment]
Thanks.  I'll update the tests accordingly.

**Comment by CyrusNajmabadi** [comment]
@gafter When after merging in the latest from your branch, and after updating the syntax as you specified (i.e. ```is``` => ```:```), i'm getting a compiler error thrown when i try to run the test:

```
  at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.GetTypeOrReturnType(Symbol symbol, RefKind& refKind, TypeSymbol& returnType, ImmutableArray`1& returnTypeCustomModifiers, ImmutableArray`1& refCustomModifiers)
   at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.GetTypeOrReturnType(Symbol symbol)
   at Microsoft.CodeAnalysis.CSharp.Binder.LookupMemberForPropertyPattern(TypeSymbol inputType, IdentifierNameSyntax name, DiagnosticBag diagnostics, Boolean& hasErrors, TypeSymbol& memberType)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindPropertySubpattern(PropertySubpatternSyntax node, TypeSymbol inputType, DiagnosticBag diagnostics, Boolean& hasErrors)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindPropertyPattern(PropertyPatternSyntax node, TypeSymbol operandType, Boolean hasErrors, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindPattern(PatternSyntax node, TypeSymbol operandType, Boolean hasErrors, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindIsPatternExpression(IsPatternExpressionSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionInternal(ExpressionSyntax node, DiagnosticBag diagnostics, Boolean invoked, Boolean indexed)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindExpression(ExpressionSyntax node, DiagnosticBag diagnostics, Boolean invoked, Boolean indexed)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindValue(ExpressionSyntax node, DiagnosticBag diagnostics, BindValueKind valueKind)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindBooleanExpression(ExpressionSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindIfStatement(IfStatementSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindStatement(StatementSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindBlockParts(BlockSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindBlock(BlockSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindStatement(StatementSyntax node, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindMethodBody(CSharpSyntaxNode declaration, BlockSyntax blockBody, ArrowExpressionClauseSyntax expressionBody, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.Binder.BindMethodBody(CSharpSyntaxNode syntax, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.BindMethodBody(MethodSymbol method, TypeCompilationState compilationState, DiagnosticBag diagnostics, ImportChain& importChain, Boolean& originalBodyNested, ValueTuple`3& forSemanticModel)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethod(MethodSymbol methodSymbol, Int32 methodOrdinal, ProcessedFieldInitializers& processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.VisitNamedType(NamedTypeSymbol symbol, TypeCompilationState arg)
   at Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol.Accept[TArgument,TResult](CSharpSymbolVisitor`2 visitor, TArgument argument)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamespace(NamespaceSymbol symbol)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethodBodies(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuiltOpt, Boolean emittingPdb, Boolean emitTestCoverageData, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnosticsForAllMethodBodies(DiagnosticBag diagnostics, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnostics(CompilationStage stage, Boolean includeEarlierStages, DiagnosticBag diagnostics, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnostics(CompilationStage stage, Boolean includeEarlierStages, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnostics(CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.<ComputeAnalyzerDiagnosticsWithoutStateTrackingAsync>d__49.MoveNext()
```

**Comment by gafter** [comment]
Will look at the test.

**Comment by gafter** [comment]
@CyrusNajmabadi I don't know what it is about your test, but I tried precisely those sources in your branch and I do see the diagnostic.

**Comment by CyrusNajmabadi** [comment]
Strange.  i will look further to see if i can figure out why i'm getting no diagnostics here.

**Comment by CyrusNajmabadi** [comment]
@gafter Figured it out.  For the switch example, if i don't have the "features" flag enabled, then the diagnostic is suppressed.  That isn't the case for the "x is Pattern" example.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide   Please review this update of "generate variable" to understand and properly generate properties when used in property-patterns.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide @jmarolf @jcouv can someone take a look at this PR?  Thanks!

File: src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs [hunk 3]
```diff
@@ -7609,5 +7614,213 @@ void Method()
...
+    {
+        object o = null;
+        if (o is Blah { X: int i })
+        {
+        }
+    }
+
+    class Blah
+    {
+        public int X { get; internal set; }
```
Comment (Neme12) [inline src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: `internal` is redundant here

File: src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs [hunk 4]
```diff
@@ -7609,5 +7614,213 @@ void Method()
...
+                break;
+        }
+    }
+
+    class Blah
+    {
+        public int X { get; internal set; }
+    }
+}", parseOptions: WithPatternsEnabled);
+        }
```
Comment (Neme12) [inline src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: consider adding tests with some error cases, eg. `Blah { X: { [|Y|]: int i } }` where X doesn't exist, and consider adding an example with more than 1 property

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: > consider adding tests with some error cases, eg. Blah { X: { [|Y|]: int i } } where X doesn't exist,
sure

> and consider adding an example with more than 1 property

It's unclear what that would be testing.  We don't generate multiple props at a time for example.

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 5]
```diff
@@ -32,28 +32,28 @@ protected override bool IsUnusableType(ITypeSymbol otherSideType)
...
                 {
                     return types;
                 }
 
-                return GetTypesComplex(expression).Where(IsUsableTypeFunc);
+                return GetTypesComplex(node).Where(IsUsableTypeFunc);
             }
 
-            private static bool DecomposeBinaryOrAssignmentExpression(ExpressionSyntax expression, out SyntaxToken operatorToken, out ExpressionSyntax left, out ExpressionSyntax right)
+            private static bool DecomposeBinaryOrAssignmentExpression(SyntaxNode node, out SyntaxToken operatorToken, out ExpressionSyntax left, out ExpressionSyntax right)
```
Comment (Neme12) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: why was this changed?

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 6]
```diff
@@ -122,10 +122,16 @@ private IEnumerable<TypeInferenceInfo> GetTypesSimple(ExpressionSyntax expressio
...
-                ExpressionSyntax expression)
+                SyntaxNode node)
             {
-                expression = expression.WalkUpParentheses();
-                var parent = expression.Parent;
+                var expression = node as ExpressionSyntax;
+                if (expression != null)
+                {
+                    expression = expression.WalkUpParentheses();
+                    node = expression;
```
Comment (Neme12) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: nit: consider pattern matching: `if (node is ExpressionSyntax expression) node = expression.WalkUpParentheses()`

File: src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs (hunk 4 above)
Comment (Neme12) [reply src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: it would be testing that this doesn't crash

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 7]
```diff
@@ -899,7 +909,7 @@ private int GetArgumentListIndex(AttributeArgumentListSyntax attributeArgumentLi
                 return (tokenIndex + 1) / 2;
             }
 
-            private IEnumerable<TypeInferenceInfo> InferTypeInBinaryOrAssignmentExpression(ExpressionSyntax binop, SyntaxToken operatorToken, ExpressionSyntax left, ExpressionSyntax right, ExpressionSyntax expressionOpt = null, SyntaxToken? previousToken = null)
+            private IEnumerable<TypeInferenceInfo> InferTypeInBinaryOrAssignmentExpression(SyntaxNode binop, SyntaxToken operatorToken, ExpressionSyntax left, ExpressionSyntax right, ExpressionSyntax expressionOpt = null, SyntaxToken? previousToken = null)
```
Comment (Neme12) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: This seems like it should still be `ExpressionSyntax`

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 8]
```diff
@@ -1420,16 +1430,83 @@ private IEnumerable<TypeInferenceInfo> InferTypeInImplicitArrayCreation(Implicit
...
+                // parent type.  So look up the parent type first, then find the X member in it
+                // and use that type.
+                if (child == subpatternElement.Pattern &&
+                    subpatternElement.NameColon.Name is IdentifierNameSyntax identifier)
+                {
+                    var parentTypes = this.InferTypes(subpatternElement);
+                    var memberTypes = 
+                        parentTypes.SelectMany(ti => GetMemberReturnTypes(ti.InferredType, identifier.Identifier.ValueText))
+                                   .SelectAsArray(t => new TypeInferenceInfo(t));
+
```
Comment (Neme12) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: why is all this logic necessary? doesn't the compiler return a symbol for X?

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 9]
```diff
@@ -1420,16 +1430,83 @@ private IEnumerable<TypeInferenceInfo> InferTypeInImplicitArrayCreation(Implicit
...
+                    return memberTypes;
+                }
+
+                return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
+            }
+
+            private ImmutableArray<ITypeSymbol> GetMemberReturnTypes(ITypeSymbol type, string memberName)
+            {
```
Comment (Neme12) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: static

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Because the type inferred now works across expressions and patterns uniformly.

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 7 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: true.  fixed.

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 5 above)
Comment (Neme12) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: I don't see how that's relevant here. This method is called DecomposeBinaryOrAssignmentExpression and still only handles expressions.

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 8 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Good question.  Let me try that.

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 9 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: done.

File: src/Features/Core/Portable/GenerateMember/AbstractGenerateMemberService.cs [hunk 10]
```diff
@@ -154,6 +154,17 @@ protected AbstractGenerateMemberService()
                 isStatic = false;
                 return;
             }
+            else if (syntaxFacts.IsNameOfSubpatternElement(expression) &&
+                     syntaxFacts.IsPropertySubpattern(expression.Parent.Parent.Parent))
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/GenerateMember/AbstractGenerateMemberService.cs]: sure.

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: sorry, i thought you were asking a general question about the use of Expression vs SyntaxNode.  I can make this ExpressionSyntax, though then i still have to add casts.  That's not a big deal, just something that wasn't necessary if this was typed as SyntaxNode.

File: src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs [hunk 11]
```diff
@@ -7609,5 +7614,213 @@ void Method()
...
+    }
+
+    class Frob
+    {
+        public int Y { get; internal set; }
+    }
+
+    class Blah
+    {
+        public Frob X;
```
Comment (Neme12) [inline src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: is there a similar test wehre X is a property rather than field?

Comment (Neme12) [reply src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs]: it would be good to have one to exercise both cases is GetMemberReturnTypes

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 8 above)
Comment (gafter) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: The semantic model is not implemented for this syntactic location yet.

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: @gafter Is the expectation that it *will* be at some later point?  I just want to know what i should expect to work in the future so i can write the feature code in the appropriate manner :)  

For example, you explained in the past how patterns don't have types, so i know to not even expect to be able to use .GetTypeInfo on a pattern.  But the rules about things like getting symbol info on these parts of a pattern are a bit less clear.

Thanks!

Comment (gafter) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Yes, I intend that identifiers referenced within a pattern will have SymbolInfo working. I will work on that next (though I am out for about a week, so it will be more than a week before I have it working).

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Sounds good.  I'll wait for that to be done.

**Comment by jaredpar** [comment]
We apologize, but we are closing this PR due to code drift. We regret letting this PR go so long without attention, but at this point the code base has changed too much for us to revisit older PRs. Going forward, we will manage PRs better under an SLA currently under draft in issue #26266 – we encourage you to add comments to that draft as you see fit.   Although that SLA is still in draft form, we nevertheless want to move forward with the identification of older PRs to give contributors as much early notice as possible to review and update them.

If you are interested in pursuing this PR, please reset it against the head of master and we will reopen it. Thank you!

**Comment by CyrusNajmabadi** [comment]
@jaredpar please reopen this PR.  This is work that @gafter has asked me to do as part of the pattern-matching feature work.  

> If you are interested in pursuing this PR, please reset it against the head of master and we will reopen it. Thank you!

This PR is intentionally against @gafter's feature branch as that's where all the pattern work is happening.

Note that work has been made in this PR as of a week ago, please don't close the PRs that are still actively being worked on, thanks :D

**Comment by jaredpar** [comment]
@CyrusNajmabadi it's apparently not just you. I can't re-open this PR either. I'm not sure what is going on here.

**Comment by sharwell** [comment]
@CyrusNajmabadi You need to create a branch **generateVarExpressions** at commit d9a8bf3 in your fork in order for this pull request to be reopened.

**Comment by CyrusNajmabadi** [comment]
@sharwell can you clarify that?  I already have a branch called generateVarExpressions .  So i can't create one.  Should i create another branch with a different name?  Should i delete the existing branch?  A set of steps would be appreciated here.

**Comment by CyrusNajmabadi** [comment]
@jaredpar Can we put a hold on closing out old PRs until a set of reasonable steps can be provided to contributers on how to actually reopen things?  Thanks!

**Comment by jaredpar** [comment]
@CyrusNajmabadi this seems to be the only PR that was affected. I've re-opened several others with no ill effects.

**Comment by CyrusNajmabadi** [comment]
Perhaps this is because it's a feature branch?  I also don't seem to be able to reopen: https://github.com/dotnet/roslyn/pull/10420#issuecomment-382805061

**Comment by CyrusNajmabadi** [comment]
I could not figure out how to reopen this.  So i've opened https://github.com/dotnet/roslyn/pull/26355 instead.
